### PR TITLE
Documentation: Update formatting and structure for Development environment document 

### PR DESCRIPTION
**Pull request by bisko**
#### Changes proposed in this Pull Request:

* This PR introduces an update to the Development Environment setup document.
* Updated:
 * Formatting
  * Some things were a bit hard to follow when I set up my environment
 * Organization
  * Some points are now grouped together 
  * Tried to order the document in more step-by-step flow, compared to more "group of notes" style

#### Testing instructions:

* Open `docs/development-environment.md`
* Read the document and see if it makes sense and does it look better than it did before

#### Proposed changelog entry for your changes:

* Not sure if this will need a changelog entry as it doesn't affect the plugin itself.

### CODE REVIEW

File: docs/development-environment.md [hunk 1]
```diff
@@ -1,298 +1,444 @@
...
-To use a custom reporter, pass the argument `-R, --reporter <name>`:
+<a name="unit-testing-php"></a>
 
-```
-$ yarn test-client -R 'my_reporter'
-```
+## PHP unit tests
 
-## Installing Composer
+* ### Docker
```
Comment (leogermani) [inline docs/development-environment.md]: The re-organization looks great to me. My only question is whether we should add this here or leave it just in the Docker page...

Comment (bisko) [reply docs/development-environment.md]: I think having the section here is good. Now that you mention it though, I'm not sure if we should be duplicating instructions between both documents, in case they change. For the moment I think this would be fine and should be updated in future iterations on the documentation.

File: docs/development-environment.md [hunk 2]
```diff
@@ -1,298 +1,444 @@
...
-To run multisite tests, run:
+	
+ You'll need to have a public SSH key setup with GitHub, which is more secure than saving your password in your keychain.
+ There are more details about [setting up a public key on GitHub.com](https://help.github.com/en/articles/adding-a-new-ssh-key-to-your-github-account).
 
-```
-phpunit -c tests/php.multisite.xml
-```
+<a name="jetpack-development-build"></a>
+## Building Jetpack
```
Comment (jeherve) [inline docs/development-environment.md]: I wonder if we should take the opportunity to split things up a bit here. We now have a build for extensions as well (we could link to https://github.com/Automattic/jetpack/blob/90a1b683c84b87a401ae707599842fe59255ab9b/extensions/README.md ), and a build for Instant Search ( https://github.com/Automattic/jetpack/blob/9aee29df9f2b1ad9031eabe5d7118ad9e5a6f73c/modules/search/instant-search/README.md )

Maybe we can mention that here?

File: docs/development-environment.md [hunk 3]
```diff
@@ -1,298 +1,444 @@
...
+	
+	```sh
+	yarn watch
+	```	
+	
+* ### Production build
+	The production build will generate minified files without duplicated code (resulting from dependencies) and will also generate the matching source map and language files. To build it use:
+	
+	```sh
+	NODE_ENV=production yarn build-client
```
Comment (jeherve) [inline docs/development-environment.md]: We can shorten that to `yarn build-production-client`.

File: docs/development-environment.md [hunk 4]
```diff
@@ -1,298 +1,444 @@
...
+	$ composer install
+	```
+	
+	This runs the actual linting task.
+	
+	```sh
+	$ composer php:lint
+	```
+	
+	_There's also a handy `yarn php:lint` that will run `composer php:lint` if you prefer_.
```
Comment (jeherve) [inline docs/development-environment.md]: I wonder if we should keep only the Yarn version in here to keep things shorter?

File: docs/development-environment.md [hunk 5]
```diff
@@ -1,298 +1,444 @@
...
+	* `wp jetpack options <list|get|delete|update> [<option_name>] [<option_value>]`
+	* `wp jetpack protect <whitelist> [<ip|ip_low-ip_high|list|clear>]`
+	* `wp jetpack reset <modules|options>`
+	* `wp jetpack disconnect <blog|user> [<user_identifier>]`
+	* `wp jetpack status`
+	* `wp jetpack status [<full>]`
+	
+	More info can be found in [our support documentation](https://jetpack.com/support/jetpack-cli/).
+
+* ### JETPACK_DEV_DEBUG
```
Comment (jeherve) [inline docs/development-environment.md]: Do we need those to be list items? Can't they just be sub-headings?

**Review by jeherve — COMMENTED** [review]
Thanks for tackling this! I left some remarks below.

File: docs/development-environment.md [hunk 6]
```diff
@@ -1,298 +1,444 @@
...
+* [Unit Testing](#unit-testing)
+   * [PHP unit testing](#unit-testing-php)
+   * [JavaScript unit testing](#unit-testing-js)
+* [Good code - linting, standards, compatibilty, etc.](#jetpack-linting-codestandards)
+	* [Coding standards](#jetpack-linting-codestandards-code-standards)
+	* [Linting](#jetpack-linting-codestandards-code-linting)
+* [Standard development & debugging tools](#devtools)
+
+      
+<a name="section-setting-up-env"></a>
```
[the remark this answers is not in the document]
Comment (bisko) [reply docs/development-environment.md]: Thanks! I removed the custom links and resorted to using GitHub's syntax. This was an adventure on it's own :)

File: docs/development-environment.md (hunk 5 above)
Comment (bisko) [reply docs/development-environment.md]: The only reason I left this as a list is to indent them. It looks a bit neater to have them indented under the headline and the list seems to be the only way to achieve that.

File: docs/development-environment.md (hunk 4 above)
Comment (bisko) [reply docs/development-environment.md]: Makes sense, as it will reduce clutter. Updated.

File: docs/development-environment.md (hunk 3 above)
Comment (bisko) [reply docs/development-environment.md]: Thanks! Updated.

File: docs/development-environment.md [hunk 7]
```diff
@@ -1,298 +1,444 @@
...
+	
+	To set up your environment with Docker, follow the [Docker environment for Jetpack Development guide](../docker/README.md).
+	
+<a name="setup-env-vvv"></a>
+
+* ### VVV
+
+	VVV is similar to Docker in how it works, but instead of setting up separate containers for the different parts it uses a single Linux virtual machine to set everything up with a nice interface.
+	
+	*...This section is still under development...*
```
[the remark this answers is not in the document]
Comment (bisko) [reply docs/development-environment.md]: Thanks! Updated to include

File: docs/development-environment.md [hunk 8]
```diff
@@ -1,298 +1,444 @@
...
+To do that you need to set up a WordPress site and give it the ability to run your local build of the Jetpack plugin code repository.
+
+There are several ways to achieve this, listed in the next section.
+
+<a name="section-running-local-jetpack"></a>
+## Running Jetpack locally
+
+To get a local WordPress site up and running you need a web server (Apache, Nginx), PHP and MySQL (or MariaDB).
+
+For easier and more organized development, there are several ways to achieve that:
```
[the remark this answers is not in the document]
Comment (bisko) [reply docs/development-environment.md]: Updated the text to mention that.

File: docs/development-environment.md (hunk 2 above)
Comment (bisko) [reply docs/development-environment.md]: Thanks! Added a section about them.

**Review by jeherve — DISMISSED** [review]
This looks good, thanks a lot for taking a pass at this. 

It should be good to merge! 👍 🚢

**Review by jeherve — APPROVED** [review]
